src/nettests/ndt.js

```diff
@@ -51,31 +43,30 @@ class NDT extends NettestBase {
 
   async run(argv) {
     await super.run(argv)
-    const args = [
-      '-o',
-      this.rawMeasurementsPath,
-      'ndt'
-    ]
-    const options = {
-      cwd: path.join(getOoniDir(), 'resources'),
-      stdio: 'pipe'
-    }
 
-    let metadata = {
-          testName: 'ndt',
-          path: this.rawMeasurementsPath
-        },
-        persist = true,
-        dbOperations = [],
-        measurementCount = 0,
+    let dbOperations = []
+
+    let measurement = Measurement.build({
+      name: 'ndt',
+      state: 'active',
+      reportFile: this.rawMeasurementsPath
+    })
+    dbOperations.push(measurement.save())
+
+    let persist = true,
         summary = {},
         progressInfo = null,
         uploadMbit = null,
         downloadMbit = null,
-        ping = null,
-        reportId = null
+        reportId = null,
+        localReportId = null
 
     const ndt = Ndt()
+
+    // XXX this is a hack while I wait for https://github.com/measurement-kit/measurement-kit-node/pull/4
+    ndt.test.set_options('no_file_report', '0')
+    ndt.test.set_output_filepath(this.rawMeasurementsPath)
```

Comment:
I do not fully understand the hackiness here, yet the PR has been merged, so you may want to update the code.

Reply:
This is actually code that I am going to kill

Reply:
Go for it! :-)